dramatiq/middleware/middleware.py

```diff
@@ -36,39 +44,39 @@ class Middleware:
     """
 
     @property
-    def actor_options(self):
+    def actor_options(self) -> set[str]:
         """The set of options that may be configured on each actor."""
         return set()
 
     @property
-    def forks(self):
+    def forks(self) -> list[Callable[[], int]]:
         """A list of functions to run in separate forks of the main
         process.
         """
         return []
 
-    def before_ack(self, broker, message):
+    def before_ack(self, broker: Broker, message: Message) -> None:
```

Comment:
I think for before/after ack/nack, it should use `MessageProxy`, rather than `Message`

Reply:
I have to dig some more through the code but at a first glance I didn’t quite find all the locations where these events are issued by the various brokers… 🤔